packages/plugins/i18n/admin/src/components/BulkLocaleActionModal.tsx

```diff
@@ -0,0 +1,258 @@
+import * as React from 'react';
+
+import { Table, useTable } from '@strapi/admin/strapi-admin';
+import { Box, Typography, IconButton, Flex, Tooltip } from '@strapi/design-system';
+import { Pencil, CheckCircle, CrossCircle, ArrowsCounterClockwise } from '@strapi/icons';
+import { DocumentStatus } from '@strapi/plugin-content-manager/strapi-admin';
+import { Modules } from '@strapi/types';
+import { stringify } from 'qs';
+import { type MessageDescriptor, useIntl } from 'react-intl';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { Locale } from '../../../shared/contracts/locales';
+import { getTranslation } from '../utils/getTranslation';
+
+import { LocaleStatus } from './CMHeaderActions';
+
+type status = Modules.Documents.Params.PublicationStatus.Kind | 'modified';
+
+/* -------------------------------------------------------------------------------------------------
+ * EntryValidationText
+ * -----------------------------------------------------------------------------------------------*/
+
+const TypographyMaxWidth = styled(Typography)`
+  max-width: 300px;
+`;
+
+interface EntryValidationTextProps {
+  status: status;
+  validationErrors?: Record<string, MessageDescriptor>;
+}
+
+const isMessageDescriptor = (obj: unknown): obj is MessageDescriptor => {
+  const possibleMessageDescriptor = obj as {
+    id?: unknown;
+    defaultMessage?: unknown;
+  };
+
+  return (
+    typeof obj === 'object' &&
+    typeof possibleMessageDescriptor.id === 'string' &&
+    typeof possibleMessageDescriptor.defaultMessage === 'string'
+  );
+};
+
+const EntryValidationText = ({ status = 'draft', validationErrors }: EntryValidationTextProps) => {
+  const { formatMessage } = useIntl();
+
+  if (validationErrors) {
+    const validationErrorsMessages = Object.entries(validationErrors)
+      .map(([key, value]) => {
+        if (isMessageDescriptor(value?.defaultMessage)) {
+          return `${key}: ${formatMessage(value.defaultMessage)}`;
+        } else if (isMessageDescriptor(value)) {
+          return `${key}: ${formatMessage(value)}`;
+        }
+
+        return `${key}: ${value}`;
+      })
+      .join(' ');
```

Comment:
I find this quite weird, how would the defaultMessage of a validation error be a MessageDescriptor?

Reply:
I think this has been improved in [`9d0b637` (#20069)](https://github.com/strapi/strapi/pull/20069/commits/9d0b63715081e6cb1ba631408b7231616c379a6d)

The issue is that the `message` property isn't always a string which doesn't match the types